### PR TITLE
Refactor `blackbox exporter` Helm chart into Golang component

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind technical-debt test
/platform all

**What this PR does / why we need it**:
This PR refactors the `blackbox exporter` and adopts the Golang component concept.

**Which issue(s) this PR fixes**:
Part of #2754

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/blackboxexporter_test.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+			blackboxExporter, err := botanist.DefaultBlackboxExporter()
+			Expect(blackboxExporter).To(BeNil())
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("#ReconcileBlackboxExporter", func() {
+		var (
+			blackboxExporter *mockblackboxexporter.MockInterface
+			kubernetesClient *kubernetesmock.MockInterface
```
Comment (shafeeqes) [inline pkg/operation/botanist/blackboxexporter_test.go]:
```suggestion
```

File: charts/shoot-core/components/charts/monitoring/charts/blackbox-exporter/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -1,10 +1,8 @@
+# TODO (acumino): drop this chart in release v1.75
```
Comment (shafeeqes) [inline charts/shoot-core/components/charts/monitoring/charts/blackbox-exporter/templates/serviceaccount.yaml]:
```suggestion
# TODO(acumino): Drop this chart in release v1.75
```

File: pkg/component/blackboxexporter/blackboxexporter.go [hunk 3]
```diff
@@ -89,7 +93,22 @@ func (b *blackboxExporter) WaitCleanup(ctx context.Context) error {
...
 		registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+
+		serviceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "blackbox-exporter",
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					v1beta1constants.GardenRole: v1beta1constants.GardenRoleMonitoring,
+					"component":                 "blackbox-exporter",
+					"origin":                    "gardener",
```
Comment (shafeeqes) [inline pkg/component/blackboxexporter/blackboxexporter.go]:
```suggestion
					managedresources.LabelKeyOrigin: managedresources.LabelValueGardener,
```

File: charts/shoot-core/components/charts/monitoring/charts/blackbox-exporter/templates/deployment.yaml [hunk 4]
```diff
@@ -1,72 +1,8 @@
-{{- define "blackbox-exporter.deployment.annotations" -}}
-reference.resources.gardener.cloud/configmap-{{ include "blackbox-exporter.config.name" . | sha256sum | trunc 8 }}: {{ include "blackbox-exporter.config.name" . }}
-{{- end -}}
----
+# TODO (acumino): drop this chart in release v1.75
```
Comment (ary1992) [inline charts/shoot-core/components/charts/monitoring/charts/blackbox-exporter/templates/deployment.yaml]:
```suggestion
# TODO(acumino): drop this chart in release v1.75
```

Same at other places too.

**Review by shafeeqes — CHANGES_REQUESTED** [review]
Thanks for the PR.
Only few minor comments.

**Review by shafeeqes — COMMENTED** [review]
/lgtm

**Pull request merged**